spark/ingestion/src/main/scala/feast/ingestion/validation/Expectation.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2022 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.ingestion.validation
+
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.{col, lit}
+import org.json4s.{CustomSerializer, DefaultFormats, Extraction, Formats, JObject, JValue}
+
+trait Expectation {
+
+  def validate: Column
+}
+
+case class ExpectColumnValuesNotToBeNull(columnName: String) extends Expectation {
```

Comment:
nit: can we rename this class to exactly match with type `expect_column_values_to_not_be_null`
ExpectColumnValuesToNotBeNull
 ?